.github/workflows/rust.yml

```diff
@@ -68,5 +68,52 @@ jobs:
       env:
         RUSTFLAGS: --cfg tokio_unstable
 
+    - name: Test
+      run: cargo test --verbose --workspace --all-features
+
+  test_linux-non-amd64:
+    name: rust (linux/${{ matrix.arch }})
+    needs: test_amd64
+    strategy:
+      matrix:
+        rust: [ "stable" ] # specify the Rust versions you want to test
+        arch: [ "arm64", "riscv64" ]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: ${{ matrix.rust }}
+        override: true
+    - name: Install QEMU
+      uses: docker/setup-qemu-action@v3
+    - name: Build wasm32-wasi Target
+      run: bash ./scripts/build_wasm_targets.sh
+      env:
+        RUSTFLAGS: --cfg tokio_unstable
+    - name: Test
+      run: cargo test --verbose --workspace --all-features
```

Comment:
I don't think this is doing anything different from on `test_amd64`. For go, we specify `GOARCH=${{ matrix.arch }} go test -v ./...` to test on a different target architecture. But here in this job I don't believe `${{ matrix.arch }}` is involved. Note `setup-qemu-action` DOES NOT CHANGE the architecture of the runner. It simply installs `binfmt` and `qemu`, which allows you to execute BINARIES build for other architectures. 

```
cargo test --verbose --workspace --all-features
```

The command above did not take in any target flag.

Reply:
Thanks for the info! I'm still experimenting these, haven't made the final version yet, might need a few more trials to make it test correctly.